frontend/src/components/accounts/AccountDetails.tsx

```diff
@@ -1,53 +1,114 @@
 import moment from "moment";
+import BN from "bn.js";
 
 import React from "react";
 
 import { Row, Col } from "react-bootstrap";
 
-import * as A from "../../libraries/explorer-wamp/accounts";
+import AccountsApi, * as A from "../../libraries/explorer-wamp/accounts";
 
 import Balance from "../utils/Balance";
 import CardCell from "../utils/CardCell";
 import TransactionLink from "../utils/TransactionLink";
 import Term from "../utils/Term";
+import AccountLink from "../utils/AccountLink";
 
 export interface Props {
   account: A.Account;
 }
 
-export default class extends React.Component<Props> {
+interface State {
+  lockupAccountId?: string;
+  lockupAmount?: string;
+  unlockupAmount?: string;
+}
+export default class extends React.Component<Props, State> {
+  state: State = {};
+
+  collectLockupInfo = async () => {
+    new AccountsApi()
+      .queryLockupAccountInfo(this.props.account.id)
+      .then((lockupInfo) => {
+        if (lockupInfo) {
+          this.setState({
+            lockupAccountId: lockupInfo.lockupAccountId,
+            lockupAmount: lockupInfo.lockupAmount,
+            unlockupAmount: lockupInfo.unlockupAmount,
+          });
+        }
+      })
+      .catch((err) => console.error(err));
+  };
```

Comment:
Move all the state manipulations away from here into `src/libraries/explorer-wamp/accounts.ts` `getAccountInfo`. This will also resolve your issue with the test.